File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/OutlineItemModel.java
package com.seafile.seadroid2.framework.model.sdoc;

import com.seafile.seadroid2.framework.model.BaseModel;

import java.util.List;

public class OutlineItemModel extends BaseModel {
    public String id;
    public String type;
    public String text;
    public boolean indent;
    public List<OutlineItemModel> children;
    public SDocDataModel data;
}

File: app/src/main/java/com/seafile/seadroid2/framework/model/sdoc/OutlineItemModel.kt
package com.seafile.seadroid2.framework.model.sdoc

import com.seafile.seadroid2.framework.model.BaseModel

class OutlineItemModel : BaseModel() {
    @JvmField
    var id: String? = null

    @JvmField
    var type: String? = null

    @JvmField
    var text: String? = null

    @JvmField
    var indent: Boolean = false

    @JvmField
    var children: MutableList<OutlineItemModel>? = null

    @JvmField
    var data: SDocDataModel? = null
}
